Ports Sign Language from /TG/Station!

JackieOTheWisp opened the pull request:
the Radio Gloves are missing though right now, if you can fix them please do, I have it commented out so they don't clutter up the inventory and be useless :\

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
Added the Tongue Tied quirk and a broken pair of Translation Gloves (which are supposed to allow you to 'speak' over comms), if you can fix this please do! I don't know honestly what I did wrong... It's been commented out of the spawn-in items meaning you don't spawn with it, so it wouldn't clutter up the inventory if you merged this like this, though I was told to save this as a draft anyways 😜 

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Another quirk can always add more fun, and from an RP standpoint this could be lots of fun!

## Changelog
:cl:
add: Added Tongue Tied trait
add: Added radio gloves!
imageadd: added sprites for the gloves, and the actual tied tongue
code: changed some code, gloves still broken though
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

JackieOTheWisp committed:
Ports Sign Language from /TG/Station!

the Radio Gloves are missing though right now, if you can fix them please do, I have it commented out so they don't clutter up the inventory and be useless :\

JackieOTheWisp commented:
fran help what's the merge conflict mean :(

Sinestia commented:
it means the file you're modifying is in a different state than what the repo has

basically the file was updated after your branch split off, likely due to this #3445 being merged

conputers try to help but it needs manually reconciled too

Archanial commented on code/_globalvars/traits.dm:
```diff
@@ -100,7 +100,8 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_NOBLOCK" = TRAIT_NOBLOCK,
-		"TRAIT_NANITECOMPATIBLE" = TRAIT_NANITECOMPATIBLE
+		"TRAIT_NANITECOMPATIBLE" = TRAIT_NANITECOMPATIBLE,
+		"TRAIT_SIGN_LANG" = TRAIT_SIGN_LANG,	
```
```suggestion
		"TRAIT_SIGN_LANG" = TRAIT_SIGN_LANG	
```

Archanial commented on code/game/objects/items/devices/radio/radio.dm:
```diff
@@ -205,6 +205,20 @@
+			if(length(empty_indexes) == 0) //Due to the requirement of gloves, the arm check for normal speech would be redundant here.
+				return FALSE
+			if(mute.handcuffed)//Would be weird if they couldn't sign but their words still went over the radio
+				return FALSE
```
```suggestion
	if(HAS_TRAIT(M, TRAIT_SIGN_LANG) && istype(M, /mob/living/carbon)) //Forces Sign Language users to wear the translation gloves to speak over radios
		var/mob/living/carbon/mute = M
		var/empty_indexes = mute.get_empty_held_indexes() //How many hands the player has empty
		var/gloves = mute.get_item_by_slot(ITEM_SLOT_GLOVES)
		if(!istype(gloves, /obj/item/clothing/gloves/radio))
			return FALSE
		if(length(empty_indexes) == 1)
			message = stars(message)
		if(length(empty_indexes) == 0) //Due to the requirement of gloves, the arm check for normal speech would be redundant here.
			return FALSE
		if(mute.handcuffed)//Would be weird if they couldn't sign but their words still went over the radio
			return FALSE
```

Archanial commented on code/modules/mob/living/say.dm:
```diff
@@ -259,6 +267,25 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	var/list/the_dead = list()
+	if(HAS_TRAIT(src, TRAIT_SIGN_LANG))
+		var/mob/living/carbon/mute = src
+		if(istype(mute))
```
```suggestion
		if(istype(src, /mob/living/carbon)
			var/mob/living/carbon/mute = src
```

Effemu commented:
wont that make mime's vow pointless?

Archanial commented:
> 
> 
> wont that make mime's vow pointless?

It won't, mimes can't use it.

JackieOTheWisp commented:
gloves still don't want to work I'm testing it on my end...

JackieOTheWisp committed:
Update code/modules/mob/living/say.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

JackieOTheWisp committed:
Update code/_globalvars/traits.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

JackieOTheWisp committed:
Update code/game/objects/items/devices/radio/radio.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

JackieOTheWisp committed:
Merge remote-tracking branch 'origin/aggressivejazzhands' into aggressivejazzhands

JackieOTheWisp committed:
Fixes from Archie, thanks! :D

The files are automatically tagging you as co-author I think but if not let me know :)

JackieOTheWisp committed:
Merge branch 'master' into aggressivejazzhands

JackieOTheWisp commented:
...Hey archie what's those three failing tests mean? I didn't do anything to the maps...

Edit: make that 4 fails 😔

Archanial commented on code/datums/traits/neutral.dm:
```diff
@@ -124,3 +124,30 @@
+.H.put_in_hands(gloves)
+H.equip_to_slot(gloves, ITEM_SLOT_GLOVES)
+H.regenerate_icons()
+
```
```suggestion
	H.put_in_hands(gloves)
	H.equip_to_slot(gloves, ITEM_SLOT_GLOVES)
	H.regenerate_icons()
```

JackieOTheWisp committed:
Update code/datums/traits/neutral.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

JackieOTheWisp committed:
Merge remote-tracking branch 'origin/aggressivejazzhands' into aggressivejazzhands

Archanial commented on code/_globalvars/traits.dm:
```diff
@@ -102,6 +102,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_NOBLOCK" = TRAIT_NOBLOCK,
 		"TRAIT_NANITECOMPATIBLE" = TRAIT_NANITECOMPATIBLE,
 		"TRAIT_WARDED" = TRAIT_WARDED
+    "TRAIT_SIGN_LANG" = TRAIT_SIGN_LANG
```
```suggestion
		"TRAIT_WARDED" = TRAIT_WARDED,
    "TRAIT_SIGN_LANG" = TRAIT_SIGN_LANG
```

JackieOTheWisp committed:
Update code/_globalvars/traits.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

JackieOTheWisp commented:
Closed because of things going on with the runechat updating, I'll retry when that gets handled

MNarath1 commented on code/modules/mob/living/say.dm:
```diff
@@ -241,6 +249,25 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	var/list/listening = get_hearers_in_view(message_range+eavesdrop_range, source)
 	var/list/the_dead = list()
+	if(HAS_TRAIT(src, TRAIT_SIGN_LANG))
+		if(istype(src, /mob/living/carbon)
```
you litterally forgot the second ) for the if the first ) is only for the istype. And yes i know i this is a closed PR
